lib/vm/src/instance/mod.rs

```diff
@@ -599,9 +597,16 @@ impl Instance {
 
         let table = self.get_table(table_index);
         let passive_elements = self.passive_elements.borrow();
-        let elem = passive_elements
-            .get(&elem_index)
-            .map_or_else(|| -> &[VMCallerCheckedAnyfunc] { &[] }, |e| &**e);
+        let elem = passive_elements.get(elem_index).map_or_else(
+            || -> &[VMCallerCheckedAnyfunc] { &[] },
+            |e| {
+                if let Some(e) = e {
+                    &**e
+                } else {
+                    &[]
+                }
+            },
+        );
```

Comment:
I wrote a long comment walking through how I simplified this and tried a few different things but the comment got deleted by Firefox / GitHub so,

```Rust
        let elem = passive_elements.get(elem_index).map_or(&[], |e| e.map_or(&[], &**e));
```

Should be fine. There's also a way where we combine the 2 options as one and only handle the failure case once, but I think it's a little less readable in this context.

`map_or` is fine because the alternative is cheap: `map_or_else` is only needed when we have an expensive to compute alternative.

Reply:
Actually I'll include it:
```Rust
        let elem = passive_elements.get(elem_index).and_then(|e| e.as_ref().map(|e| &**e)).unwrap_or(&[]);
```

In case you're interested. It may be able to be simplified more with `as_deref`

maybe

```Rust
        let elem = passive_elements.get(elem_index).and_then(|e| e.as_ref().as_deref()).unwrap_or(&[]);
```

Reply:
I went to the second suggestion since it only has one `&[]`! Thanks!